dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java

```diff
@@ -130,24 +130,26 @@ public static PageResourceHelper getInstance() {
         return PageResourceHelper.SingletonHolder.INSTANCE;
     }
 
-    public Template saveTemplate(final User user, IHTMLPage htmlPageAsset, final PageForm pageForm)
+    public HTMLPageAsset saveTemplate(final User user, HTMLPageAsset htmlPageAsset, final PageForm pageForm)
 
             throws BadRequestException, DotDataException, DotSecurityException {
 
         try {
-            
-            
             Template templateSaved = this.saveTemplate(htmlPageAsset, user, pageForm);
 
             String templateId = htmlPageAsset.getTemplateId();
 
+            Contentlet contentlet = htmlPageAsset;
+
             if (!templateId.equals( templateSaved.getIdentifier() )) {
-                htmlPageAsset.setInode(null);
-                htmlPageAsset.setTemplateId(templateSaved.getIdentifier());
-                this.contentletAPI.checkin((HTMLPageAsset)htmlPageAsset, user, false);
+                contentlet = this.contentletAPI.checkout(htmlPageAsset.getInode(), user, false);
+                contentlet.setStringProperty(HTMLPageAssetAPI.TEMPLATE_FIELD, templateSaved.getIdentifier());
+                contentlet = this.contentletAPI.checkin(contentlet, user, false);
```

Comment:
You might want to db commit after this so the change is visible instantly to the next rest request

Reply:
I added @WrapInTransaction annotation